etc/serialization/pre-commit

```diff
@@ -0,0 +1,238 @@
+#!/bin/bash
+#
+# pre-commit hook script that activates the Sesame serializer to serialize any OWL files in a standard format just before
+# they are committed to the (local) Github repository.
+#
+# Called by "git commit" with no arguments. This hook can stop the commit when it encounters an error.
+#
+# To enable this hook, rename this file to "pre-commit" and save it in the directory <your git clone root>/.git/hooks
+#
+
+# Customized by the git user
+export JAVA_HOME= location of your Java goes here
+
+#
+# Redirect all output generated by this script to stderr.
+#
+exec 1>&2
+
+log_prefix="rdf-toolkit: sesame-serializer: "
+
+function log() {
+  echo "${log_prefix}$@"
+}
+
+function log_error() {
+  log "ERROR: $@"
+}
+
+log "This is the pre-commit hook"
+
+java_exe=""
+base_dir=""
+
+function findJava() {
+
+  whichJava=""
+
+  java_home=""
+
+  if [ "${RDF_TOOLKIT_JAVA_HOME}" != "" ] ; then
+    java_home="${RDF_TOOLKIT_JAVA_HOME}"
+  fi
+  if [ "${JAVA_HOME}" != "" ] ; then
+    java_home="${JAVA_HOME}"
+  fi
+  if [ "${java_home}" == "" ] ; then
+      log_error A-$JAVA_HOME
+    log_error "Please set RDF_TOOLKIT_JAVA_HOME or JAVA_HOME to point to a Java 7 or later installation."
+    return 1
+  fi
+  java_home=${java_home/C:\\/\/c\/}
+  java_home=${java_home//\\/\/}
+  log "java_home =" ${java_home}
+
+  if [ -x "${java_home}/bin/java" ] ; then
+    whichJava="${java_home}/bin/java"
+  else
+    log_error "Could not find java in your RDF_TOOLKIT_JAVA_HOME or JAVA_HOME: ${java_home}"
+    log_error "Please set RDF_TOOLKIT_JAVA_HOME or JAVA_HOME to point to a Java 7 or later installation."
+    return 1
+  fi
+  log "whichJava =" $whichJava
+
+  local versionJava=$( "$whichJava" -version 2>&1 | head -n 1 | cut -d \"  -f 2 )
+  log "versionJava = " $versionJava
+
+  set -- ${versionJava//./ }
+
+  local java_major=$1
+  local java_minor=$2
+
+  if [ $java_major -eq 1 -a $java_minor -gt 7 ] ; then
```

Comment:
Stupid test requires Java version of 1.x.

Reply:
I don't remember where this came from; is there a correct test we can make here?